src/main/scala/org/broadinstitute/clio/dataaccess/CachedServerStatusDAO.scala

```diff
@@ -0,0 +1,47 @@
+package org.broadinstitute.clio.dataaccess
+
+import org.broadinstitute.clio.model.ServerStatusInfo
+
+import scala.concurrent.Future
+import scala.util.Try
+
+/**
+  * Stores the server status in a synchronized mutable variable.
+  */
+class CachedServerStatusDAO private() extends ServerStatusDAO {
+  private val mutex = new Object()
+  private var currentStatus: ServerStatusInfo = ServerStatusInfo.NotStarted
+
+  override def setStatus(status: ServerStatusInfo): Future[Unit] = Future.fromTry(
+    Try {
+      mutex synchronized {
+        import CachedServerStatusDAO.level
+        if (level(currentStatus) < level(status))
+          currentStatus = status
+      }
+    }
+  )
+
+  override def getStatus: Future[ServerStatusInfo] = Future.fromTry(
+    Try {
+      mutex synchronized currentStatus
+    }
+  )
+}
+
+object CachedServerStatusDAO {
+  def apply(): CachedServerStatusDAO = {
+    new CachedServerStatusDAO()
+  }
+
+  private def level(status: ServerStatusInfo): Int = {
+    status match {
+      case ServerStatusInfo.NotStarted => 0
```

Comment:
I wrestled with this one a bit. Ultimately I decided to follow the firecloud-orchestration [DEVNOTES](https://github.com/broadinstitute/firecloud-orchestration/blob/63dcb0be65b2fe69234b0e139c4ce6e57bf90087/DEVNOTES.md), that are copied into the clio [wiki](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509#Clio(MetadataManager)-PackageStructureandSingleResponsibilities).

> - model
>     -  model classes. Case classes if possible. Little-to-no logic implementations; things like toString allowed

Thus the `ServerStatusInfo` model objects passed around do not contain any business logic, not even the ability to transition from one "level" to another.